test/integration/spec/localparticipant.js

```diff
@@ -1394,4 +1394,48 @@ describe('LocalParticipant', function() {
       assert(!error);
     });
   });
+
+  describe.only('JSDK-2985 - Event Listeners cleaned up after leaving room', () => {
+    let room;
+    let error;
+    let listenerCount = 0;
+    const events = ['enabled', 'disabled', 'started', 'stopped', 'message'];
+    async function setupRoom(token, options) {
+      let room = await connect(token, options);
+      return room;
+    }
+
+    before(async () => {
+      try {
+        const token = getToken(randomName());
+        const options = Object.assign({ audio: true, fake: true }, defaults);
+        // eslint-disable-next-line no-await-in-loop
+        room = await setupRoom(token, options);
+        if (room) {
+          room.disconnect();
+        }
+        events.forEach(event => {
+          room.localParticipant.tracks.forEach(publication => {
+            listenerCount += publication.track.listenerCount(event);
```

Comment:
Looks like a good test - does this test fail w/o your fix ? How about adding multiple connect/disconnect in this test ?

Reply:
Yes! I took out the extra connects/disconnects because I realized I only wanted to see if the listeners were being cleaned up after we leave the room. Without the fix, the test fails and catches 7 listeners. With the fix, it catches 0 listeners and the test passes.